### PR TITLE
Fixed Log in, Created Redux slice for events. Improved View Profile P…

### PR DESCRIPTION
**Pull request by NeelSalvi**
#193 #195

### CODE REVIEW

File: client/TrailSeek/src/app/userSlice.js [hunk 1]
```diff
@@ -67,6 +72,27 @@ export const signUp = createAsyncThunk(
   }
 );
 
+export const fetchUserData = createAsyncThunk(
+  "user/fetchUserData",
+  async (_, { rejectWithValue, getState }) => {
+    try {
+      const response = await trailSeek.get("/user", {
+        headers: { Authorization: `Bearer ${getState().user.profile.token}` },
```
Comment (viniciusvieir) [inline client/TrailSeek/src/app/userSlice.js]: Is it possible to move this to the trailSeek API in /src/api/trailSeek.js so we don't have to always set the authorisation token manually?

**Review by viniciusvieir — CHANGES_REQUESTED** [review]

File: client/TrailSeek/src/app/userSlice.js (hunk 1 above)
Comment (NeelSalvi) [reply client/TrailSeek/src/app/userSlice.js]: Right now that file is used for all the trails and other forms of requests. I can probably do it. Will look into it.

Comment (viniciusvieir) [reply client/TrailSeek/src/app/userSlice.js]: The idea is to have all the headers set automatically whenever a request is made to the trailseek APIs. Maybe one way to do it is instantiate the API with the state [getState()]. Just an idea, not sure if that would work for all cases.

File: client/TrailSeek/src/api/trailSeek.js [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+    if (value) {
+      token = value;
+    }
+  } catch (error) {
+    console.log(error.response);
+  }
+  config.headers.Authorization = token ? `Bearer ${token}` : "";
+  return config;
+});
+
```
Comment (NeelSalvi) [inline client/TrailSeek/src/api/trailSeek.js:21]: Added as requested @viniciusvieir

**Review by NeelSalvi — COMMENTED** [review]
Changed

**Comment by viniciusvieir** [comment]
Looks good, thanks

**Pull request merged**